File: implementations/react/src/patterns/behavioral/command/remove-item-command.ts
import type { Command } from "./command-interface";
import type { Receiver } from "./receiver";

/**
 * Comando concreto para eliminar un ítem en el patrón Command.
 */
export class RemoveItemCommand implements Command {
    private device: Receiver;

    constructor(device: Receiver){
        this.device = device;
    }

    execute(item: string) {
        this.device.removeItem(item);
    }
}

File: implementations/flutter/design_patterns_flutter/lib/patterns/behavioral/command/remove_item_command.dart
import 'package:design_patterns_flutter/patterns/behavioral/command/command.dart';
import 'package:design_patterns_flutter/patterns/behavioral/command/receiver.dart';

class RemoveItemCommand implements Command {
  final Receiver device;
  RemoveItemCommand(this.device);

  @override
  void execute([String? item]) {
    if (item != null && item.isNotEmpty) {
      device.removeItem(item);
    }
  }
}
